src/includes/class-alma-wc-payment-upon-trigger.php

```diff
@@ -0,0 +1,198 @@
+<?php
+/**
+ * Alma Payment Upon Trigger
+ *
+ * @package Alma_WooCommerce_Gateway
+ */
+
+use Alma\API\RequestError;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	die( 'Not allowed' ); // Exit if accessed directly.
+}
+
+/**
+ * Alma_WC_Payment_Upon_Trigger
+ */
+class Alma_WC_Payment_Upon_Trigger {
+
+	/**
+	 * Logger
+	 *
+	 * @var Alma_WC_Logger
+	 */
+	private $logger;
+
+	/**
+	 * __construct.
+	 */
+	public function __construct() {
+		$this->logger = new Alma_WC_Logger();
+		add_action( 'woocommerce_order_status_changed', array( $this, 'woocommerce_order_status_changed' ), 10, 3 );
+	}
+
+	/**
+	 * Callback function for the event "order status changed".
+	 *
+	 * @param integer $order_id The order id.
+	 * @param string  $previous_status Order status before it changes.
+	 * @param string  $next_status Order status affected to the order.
+	 * @return void
+	 */
+	public function woocommerce_order_status_changed( $order_id, $previous_status, $next_status ) { // phpcs:ignore Generic.CodeAnalysis.UnusedFunctionParameter
+
+		$order = wc_get_order( $order_id );
+		if ( 'alma' !== $order->get_payment_method() ) {
+			return;
+		}
+
+		if ( 'yes' !== alma_wc_plugin()->settings->payment_upon_trigger_enabled ) {
+			return;
+		}
+
+		if ( alma_wc_plugin()->settings->payment_upon_trigger_event === $next_status ) {
+			$this->trigger_payment( $order_id, $next_status );
+		}
+	}
+
+	/**
+	 * Launches the payment on trigger for an order.
+	 *
+	 * @param integer $order_id The order id.
+	 * @return void
+	 */
+	private function trigger_payment( $order_id, $next_status ) {
+
+		$alma = alma_wc_plugin()->get_alma_client();
+		if ( ! $alma ) {
+			return;
+		}
+
+        $order = wc_get_order( $order_id );
+		if ( ! $order->get_transaction_id() ) {
+			$this->logger->error( 'Error while getting transaction_id for order_id = ' . $order_id );
+			return;
+		}
+
+		$payment = $alma->payments->fetch( $order->get_transaction_id() );
+		if ( $payment->deferred_trigger_applied ) {
+			$this->logger->info( 'Order number ' . $order_id . ' has already a value : -->' . deferred_trigger_applied . '<--.' );
+			return;
+		}
+
+		try {
+			$alma->payments->trigger( $order->get_transaction_id() );
+            $order->add_order_note( sprintf( __( 'The first customer payment has been triggered, as you updated the order status to "%s".', 'alma-woocommerce-gateway' ), $next_status ) );
```

Comment:
You can remove this note because it will be odd to manage note only for trigger payment and not for the basic create payment behavior.

further, if we add a note when everything is ok, we have to add note when it is not ok... so, this feature is not for today.

Just remove this `add_order_note` and the `$next_status` parameter for the moment